File: sdks/php/src/Resource/Sms.php
<?php

declare(strict_types=1);

namespace Pushfy\Resource;

use Pushfy\Pushfy;

/** SMS sending. */
final class Sms
{
    /** @var Pushfy */
    private $client;

    public function __construct(Pushfy $client)
    {
        $this->client = $client;
    }

    /**
     * Send a single SMS.
     *
     * @param array{to: string, text: string, extId?: string} $params
     * @return mixed The API result array [{ id, phone, date, ext_id }].
     */
    public function send(array $params)
    {
        return $this->client->classicRequest('POST', '/webapi', [
            'json' => ['messages' => [Message::normalize($params)]],
        ]);
    }

    /**
     * Send many SMS in one request.
     *
     * @param array<int, array{to: string, text: string, extId?: string}> $list
     * @return mixed
     */
    public function sendBulk(array $list)
    {
        $messages = array_map([Message::class, 'normalize'], $list);
        return $this->client->classicRequest('POST', '/webapi', [
            'json' => ['messages' => $messages],
        ]);
    }
}

File: sdks/go/sms.go
package pushfy

import "context"

// SMSResource sends SMS messages via the Messaging API.
type SMSResource struct{ c *Client }

// SMSMessage is a single SMS to send. To is normalized to digits only.
type SMSMessage struct {
	To    string
	Text  string
	ExtID string
}

func (m SMSMessage) toAPI() apiMessage {
	return apiMessage{
		Destinations: []apiDest{{To: onlyDigits(m.To)}},
		Text:         m.Text,
		ExtID:        m.ExtID,
	}
}

// Send sends a single SMS and returns the accepted messages.
func (r *SMSResource) Send(ctx context.Context, m SMSMessage) ([]MessageResult, error) {
	return r.SendBulk(ctx, []SMSMessage{m})
}

// SendBulk sends many SMS in one request.
func (r *SMSResource) SendBulk(ctx context.Context, list []SMSMessage) ([]MessageResult, error) {
	msgs := make([]apiMessage, len(list))
	for i, m := range list {
		msgs[i] = m.toAPI()
	}
	var out []MessageResult
	err := r.c.classic(ctx, "POST", "/webapi", classicOpts{json: messagesEnvelope{Messages: msgs}}, &out)
	return out, err
}
